File: examples/typescript/react/notes-showcase/src/models/noteModel.ts
/**
 * NoteModel — pure-data record for a single note.
 *
 * Immutable. `tags` is a readonly tuple-like array. `createdAt` / `updatedAt`
 * are ISO-8601 strings (mirrors the C# DateTimeOffset round-trip format and
 * the Python isoformat representation — same shape across all three flavors).
 */
export interface NoteModel {
  readonly id: string;
  readonly notebookId: string;
  readonly title: string;
  readonly tags: readonly string[];
  readonly body: string;
  readonly starred: boolean;
  readonly createdAt: string;
  readonly updatedAt: string;
}

File: examples/python/textual/notes_showcase/src/notes_showcase/models/note_model.py
"""NoteModel — pure-data record for a single note."""

from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True, slots=True)
class NoteModel:
    """Immutable note record.

    ``tags`` is a tuple so the dataclass remains hashable and value-equality
    is structurally honest. ``created_at`` / ``updated_at`` are timezone-aware
    UTC ``datetime`` objects (mirrors the C# ``DateTimeOffset``).
    """

    id: str
    notebook_id: str
    title: str
    tags: tuple[str, ...]
    body: str
    starred: bool
    created_at: datetime
    updated_at: datetime
